docs/quickstart.mdx

```diff
@@ -152,6 +164,23 @@ svixClient.Message.Create("app_Xzx8bQeOB1D1XEYmAJaRGoj0", &svix.MessageIn{
 })
 ```
 
+</TabItem>
+<TabItem value="java">
+
+```java
+Svix svix = new Svix("AUTH_TOKEN");
+svix.getMessage().create("app_Xzx8bQeOB1D1XEYmAJaRGoj0", 
+  new MessageIn()
+    .eventType("invoice.paid")
+    .eventId("evt_Wqb1k73rXprtTm7Qdlr38G")
+    .payload("{
+       \"id\": \"invoice_WF7WtCLFFtd8ubcTgboSFNql\",
+       \"status\":  \"paid\",
+       \"attempt\": 2
+      }"));
```

Comment:
Payload takes an object not a string (are multiline strings like this valid in java?)

Reply:
It's still fine to pass a string to payload (as it's an object).

Re multiline: it's not valid, I thought it will be clearer but that's a good point. Fixed

Reply:
> It's still fine to pass a string to payload (as it's an object).

Oh interesting! That's good to know.

After running some tests now that I know this it looks like if we pass a string it gets passed directly to the api as a string (no json parsing) so the api responds:
request:
```json
{"eventType":"user.created","payload":"{username: \"test_user\", email: \"test@example.com\"}"}
```
response:
```json
{"detail":[{"loc":["body","payload"],"msg":"value is not a valid dict","type":"type_error.dict"}]}
```



Interestingly passing a HashMap results in no payload:
request:
```json
{"eventType":"user.created"}
```
response:
```json
{"detail":[{"loc":["body","payload"],"msg":"value is not a valid dict","type":"type_error.dict"}]}
```



The only thing that seems to work is an actual class. 
```java
class Payload {
    public String username;
    public String email;

    public Payload(String username, String email){
        this.username = username;
        this.email = email;
    }
}
```
request:
```json
{"eventType":"user.created","payload":{"username":"new_user","email":"new_user@example.com"}}
```
Success!

This makes me pretty unhappy with the openapi generated code.  IMO don't think this is something that is worth us spending time making nicer right now but we should really revisit the Java library at a later date.  For now I think it's probably sufficient to just show an example passing a class.

Reply:
I saw in the MessageIn doc that something like this should be valid:
`@ApiModelProperty(example = "{\"username\":\"test_user\",\"email\":\"test@example.com\"}", required = true, value = "")`

In your example you don't have quotes around the JSON keys 
> `{"eventType":"user.created","payload":"{username: \"test_user\", email: \"test@example.com\"}"}`
I just wanna double check that this isn't the issue here.

If that doesn't work, I think it will make sense to wrap MessageIn with our own model to properly handle it, or at least document it somewhere and I'll change the example here to use JSON.class or something

Reply:
Interesting a that a class works. We should be explicit about it. I wonder if there's a shorter/easier way of passing hashmap too, or defining classes inline.

Reply:
Sorry, I did run with a properly quoted string which produces the same invalid request: `{"eventType":"user.created","payload":"{\"username\":\"test_user\",\"email\":\"test@example.com\"}"}`
I'll try to dig more into why this doesn't work.

Reply:
Passing a string is wrong, because it passes payload as a string instead of an object. It can't work the way it is, no matter how you quote things.

Reply:
That's annoying, so should we just use an object in the example? 
Do we want to support strings?

Reply:
How is the code actually doing the serialization? Is there a standard java interface? What's going on below the hood?

Reply:
When setting `payload` on `MessageIn` It just gets set a field of type `Object`, the entire `MessageIn` object eventually gets passed to [`gson.toJson(obj)`](https://github.com/google/gson) and since it's of type string it just keeps it as is. Not familiar with gson so I'm not yet sure why the HashMap just disappears, but looking into it whether or not we can configure this behavior.  OpenAPI-Generator also has a few options for serialization library so if gson doesn't have any way around this its possible one of the others may.

Reply:
Worth figuring it out. I think we should probably support obj and hashmap, and probably complain about string.

Reply:
ah so it converts the MessageIn object to json, I think that the fact that it doesn't convert the string to a  json is working as intended. I'm surprised though that HashMap doesn't work, gson does support it.

Maybe we should try to serialise the payload in Message.create?
Something like this:
`
public MessageOut create(final String appId, final MessageIn messageIn) throws ApiException {
  return api.createMessageApiV1AppAppIdMsgPost(appId, 
                 messageIn.payload(new JSON().serialise(messageIn.getPayload())));
}
`

Reply:
Since this is the only endpoint that currently takes in arbitrary JSON structures I think doing our own validation here (throw on non-json string) could be a good stopgap solution. I'll take a look and see if parsing here solves the issue I ran into with HashMaps as well.

Rather than serializing, we'd want to check if it was a string, try to deserialize it into something that the next pass of json serialization will handle properly though right?

Reply:
I think it depends on whether HashMap works or not, if this is the way to also fix map I would just convert it to a JSONObject in that method. 
We will need to revisit it once we have other endpoints or if we would want to accept Strings (non-JSON)

Reply:
I'm all in favour of making it String only for now. So the user of the library will have to implement their own serialization. After that we can figure out things like hashmaps and classes.

Reply:
In all languages or only Java? We can verify for now that the payload is a String. Should we require any format or let the user do what they want?

Reply:
Only Java, because only here it's complicated. :)

It should be a valid JSON string. Are we able to achieve that?

Reply:
I can send a PR that tries to convert the payload to a jsonobject before sending the message, I think it will fix all cases, I'll test it also with Hashmap.
If we want, we can also support non-json Strings object

Reply:
Please! If this fixes things, that's amazing. Otherwise, even just supporting String and nothing else is fine. Though again, the string needs to become a real json object. We can't send `payload: "{\"foo\":1}"`. That's wrong.

Reply:
Frank, I created https://github.com/svix/svix-libs/pull/57 to handle strings. 
I think it's worth adding a test file for this use case though I'm not sure where it fits. I can create MessageTest for the class and do the conversion, it's not ideal but will cover it. wdyt?
In addition, I'm still looking into the hashmap, the weird thing is that when I try it with the following syntax it works:

`HashMap<String, String> map = new HashMap<String, String>();
		map.put("name", "new_user");
		map.put("email", "new_user@email.com");
`